validator/validate_api.py

```diff
@@ -72,21 +66,28 @@ def setup_parse_and_data(setup_state):
 
 def get_question_data_by_key(key, val):
     df = current_app.df
-    first_q = df["questions"][df["questions"][key] == val].iloc[0]
-    module_id = first_q.cvuid
-    uid = first_q.uid
-    has_numeric = df["questions"][df["questions"][key] == val].iloc[0].contains_number
+    # FIXME - should use all the questions and combine associated pages
+    # FIXME - last_q works better because of some dirty data getting through
+    # that has innovation pages but not the exact book those pages are from
+    last_q = df["questions"][df["questions"][key] == val].iloc[-1]
+    module_id = last_q.cvuid
```

Comment:
I am interested in what logic you are using to get this.
Tutor finds all pages in the current book that matches the tags, then takes the last page in ToC order as the page the question will be assigned to.
Hopefully response-validator does the same?

Reply:
Nope, there's no inherent order  to these dataframes, as far as I can tell. Just found that last worked better than first, in the presence of some missing data. Plan to revisit this w/ an explicit ordering at some point, hence the FIXMEs :/